### PR TITLE
Add Exception Manager

### PR DESCRIPTION
**Pull request by NickGrantONS**
# Motivation and Context
We need an exception manager to help us view and skip unprocessable messages with zero downtime.

# What has changed
Added the Exception Manager.

# How to test?
Build. Run. Publish bad message. Check the logs. Check the Exception Manager.

# Links
Trello: https://trello.com/c/2rkYNAMa

### CODE REVIEW

**Comment by NickGrantONS** [comment]
[Exception Manager v1.0](https://trello.com/c/2rkYNAMa/1266-exception-manager-v10)

File: src/main/java/uk/gov/ons/census/action/messaging/ManagedMessageRecoverer.java [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+    } catch (Exception exceptionManagerClientException) {
+      log.with("message_hash", messageHash)
+          .warn(
+              "Unable to store a copy of the message. Will NOT be skipping",
+              exceptionManagerClientException);
+    }
+
+    // Check if OK and the message is stored... then we can go ahead and quarantine
+    if (result) {
+      result = false; // The next bit might go wrong
```
Comment (ryangrundy7) [inline src/main/java/uk/gov/ons/census/action/messaging/ManagedMessageRecoverer.java]: How could it go wrong?

Comment (NickGrantONS) [reply src/main/java/uk/gov/ons/census/action/messaging/ManagedMessageRecoverer.java]: just a misleading comment really

File: src/main/java/uk/gov/ons/census/action/builders/QidUacBuilder.java [hunk 2]
```diff
@@ -190,8 +190,8 @@ public static String calculateQuestionnaireType(String treatmentCode) {
           return "34";
       }
     } else {
-      log.with("treatment_code", treatmentCode).error(UNEXPECTED_CASE_TYPE_ERROR);
-      throw new IllegalArgumentException();
+      throw new IllegalArgumentException(
+          String.format("Unexpected Case Type fortreatment code '%s'", treatmentCode));
```
Comment (hbrace) [inline src/main/java/uk/gov/ons/census/action/builders/QidUacBuilder.java]: Needs a space between for and treatment

Comment (NickGrantONS) [reply src/main/java/uk/gov/ons/census/action/builders/QidUacBuilder.java]: fixed

**Pull request merged**